### ISSUE TITLE
Option to restart player

### ISSUE THREAD

**Issue by Ro-Den** [streamlink/streamlink#1816]
### Checklist

- [ ] This is a bug report.
- [x] This is a feature request.
- [ ] This is a plugin (improvement) request.
- [ ] I have read the contribution guidelines.

### Description

I use VLC player and sometimes it immediately crashes before playing a stream.

Is there any option to restart a player without restarting Streamlink?

Or is it possible to add an option like `--player-restart SECONDS` to automatically restart a video player if it was closed within the specified period?

[Love Streamlink? Please `consider` supporting our collective. `Thanks!](https://opencollective.com/streamlink/donate)`

**Comment by Ro-Den** [comment]
@gravyboat

No, VLC player doesn't support any kind of restart function. I think it's not so easy to find the cause of crashing. It crashes randomly. Maybe, the problem is on the server side.

@bastimeyer 

Other stream output methods would cause other problems. It's not so difficult to restart Streamlink manually. The idea is to automate the proccess.

So, I added/changed the following in the main.py:

`import time`

...

def read_stream(stream, output, prebuffer, chunk_size=8192):

...

    start_time = time.time()

...

            except IOError as err:

...

                duration = time.time() - start_time
                if duration <= 3:
                    handle_url()
                else:
                    break

It works restarting a plugin. So, the final question. Is it possible to do the same without restarting a plugin?

**Comment by beardypig** [comment]
@Ro-Den, maybe you could call `output_stream(stream)` instead - it's not very clean to do either `output_stream` or `handle_url` though :)

When VLC crashes, what log message do you get from streamlink? `Player closed` or `Error when reading from stream`?

**Comment by Ro-Den** [comment]
@beardypig, VLC always crashes with `Player closed: [Errno 32] Broken pipe`.

`output_stream(stream)` is not working:

        prebuffer = stream_fd.read(8192)
    AttributeError: 'NoneType' object has no attribute 'read'

As I understand, when a video player gets closed, Streamlink automatically shuts down stream output. And it is not so easy to amend. So, `handle_url()` is the only option in this case, of course as a temporary solution.
